storagenode/piecestore/endpoint.go

```diff
@@ -64,6 +64,10 @@ type Config struct {
 	ReportCapacityThreshold memory.Size   `help:"threshold below which to immediately notify satellite of capacity" default:"500MB" hidden:"true"`
 	MaxUsedSerialsSize      memory.Size   `help:"amount of memory allowed for used serials store - once surpassed, serials will be dropped at random" default:"1MB"`
 
+	MinUploadSpeed                    memory.Size   `help:"a client upload speed should not be lower than MinUploadSpeed in bytes-per-second (E.g: 1Mb), otherwise, it will be flagged as slow-connection and potentially be closed" default:"0Mb"`
+	MinUploadSpeedGraceDuration       time.Duration `help:"if MinUploadSpeed is configured, after a period of time after the client initiated the upload, the server will flag unusually slow upload client" default:"0h0m10s"`
+	MinUploadSpeedCongestionThreshold float64       `help:"defines the proportion of MaxConcurrentRequests that must be reached before the storage node will begin dropping slow upload connections to preserve its own bandwidth" default:"0.8"`
```

Comment:
I think this is backwards, now. The storage node will drop slow upload connections _until_ this proportion is reached, with the logic in `(*speedEstimation).EnsureLimit()`.